lib/broccoli/ember-app.js

```diff
@@ -1397,11 +1401,21 @@ class EmberApp {
         throw new SilentError(`Style file cannot have the name of the application - ${this.name}`);
       }
 
+      let directory = '/app/styles';
+
+      if (experiments.MODULE_UNIFICATION) {
```

Comment:
This is the main piece of code required to support preprocessor support within the `src/ui/styles` directory.

Without this check passing in the styles directory tree path the preprocessors (at least when tested with sass) would get `app/styles` hardcoded.

The `if` statements here are to check that module unification is enabled and that the current project has styles located in the `/src/ui/styles` directory (adopted logic from the `_initOptions` method).